editor/src/document/properties_panel_message_handler.rs

```diff
@@ -258,16 +258,16 @@ fn register_layer_properties(layer: &Layer, responses: &mut VecDeque<Message>) {
 			vec![]
 		}
 		LayerDataType::Shape(shape) => {
-			vec![
-				node_section_transform(layer),
-				node_section_fill(&shape.style.fill()),
-				node_section_stroke(&shape.style.stroke().unwrap_or_default()),
-			]
+			if let Some(fill_layout) = node_section_fill(&shape.style.fill()) {
+				vec![node_section_transform(layer), fill_layout, node_section_stroke(&shape.style.stroke().unwrap_or_default())]
+			} else {
+				vec![node_section_transform(layer), node_section_stroke(&shape.style.stroke().unwrap_or_default())]
+			}
 		}
 		LayerDataType::Text(text) => {
 			vec![
 				node_section_transform(layer),
-				node_section_fill(&text.style.fill()),
+				node_section_fill(&text.style.fill()).expect("Text should have fill"),
```

Comment:
Why should text always have fill? Or does that mean something else (which could perhaps be clarified)?